### PR TITLE
Simpler

### PR DESCRIPTION
**Pull request by josandmann**
Contains fast ia-effect sampling procedure.
sample_ia_effects_window.py uses environment variables
sample_ia_effects_window_cmd.py can be run from the command line.
sampling_utils.py contains sped-up and optimized sampling utilities
requirements.txt has been updated to include numba

### CODE REVIEW

**Comment by pnieters** [comment]
@kreuzert : Jonas included two different, callable versions of the IA sampler. One using the SGE_[...] environment variable (1-100 for 100 IA samplers, but we had trouble using this before on JSC), the other one using a straight forward command line argument for the ID.
Can't we delete the SGE_GRID_ID variant and only merge the version using the command line argument?

File: src/sampling_utils.py [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+
+                    ######## x_all ########
+
+                    # calc random location-id for each sample
+                    rnd_locid_per_smpl = np.floor( av_locs_per_smpl * rnd_loc.random((n_total,)) ).astype("int32")
+
+                    # collect locations for each sample with its random location-id
+                    x_all = [ locations_by_county_np[c_offs+j][rnd_locid_per_smpl[(i*n_counties+j)*num_tps+x]] for i in range(n_days) for j in range(n_counties) for x in range(num_tps) ]
+
+                    return t_all, x_all
```
Comment (pnieters) [inline src/sampling_utils.py]: too many tabs! please try to keep to a 90 characters per line limit where possible!

**Review by pnieters — CHANGES_REQUESTED** [review]
some formatting changes required;
i will get back to you after feedback from Tim w.r.t 2 different files to start the ia_sampler!

File: src/sampling_utils.py (hunk 1 above)
Comment (josandmann) [reply src/sampling_utils.py]: Fixed it. Lines now should not exceed 60 characters

**Review by pnieters — CHANGES_REQUESTED** [review]
Please remove the SGE_DATE_ID version, rename the cmd line argument version to sample_ia_effects.py. Tim and I agree, that the env vars variant will only cause future problems!

**Comment by josandmann** [comment]
SGE_DATE_ID-version removed, command line version renamed to "sample_ia_effects.py"

**Comment by pnieters** [comment]
Let's give it a-go :+1:

**Pull request merged**